### PR TITLE
SDD-920 create delete unpublished course details page

### PR DESCRIPTION
**Pull request by MrKevJoy**
- Created Handler to mark course row upload as deleted.
- Added Tests
- New page to delete course row.

### CODE REVIEW

File: src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.Courses.cs [hunk 1]
```diff
@@ -19,6 +19,34 @@ namespace Dfc.CourseDirectory.Core.DataManagement
 {
     public partial class FileUploadProcessor
     {
+        public async Task<bool> DeleteCourseUploadRowForProvider(Guid providerId, int rowNumber)
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.Courses.cs:22]: Nit: whitespace is a little wonky in this method

File: src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.Courses.cs [hunk 2]
```diff
@@ -19,6 +19,34 @@ namespace Dfc.CourseDirectory.Core.DataManagement
...
+        {
+            using (var dispatcher = _sqlQueryDispatcherFactory.CreateDispatcher())
+            {
+                var courseUpload = await dispatcher.ExecuteQuery(new GetLatestUnpublishedCourseUploadForProvider()
+                {
+                    ProviderId = providerId
+                });
+                if (courseUpload == null)
+                    return false;
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.Courses.cs:31]: I think this should throw:
```
throw new InvalidStateException(InvalidStateReason.NoUnpublishedVenueUpload);
```

File: src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs [hunk 3]
```diff
@@ -106,6 +106,14 @@ public static ParsedCsvCourseRow FromCsvCourseRow(CsvCourseRow row, IEnumerable<
             _ => (CourseStudyMode?)null
         };
 
+        public static string ResolveDeliveryMode(CourseDeliveryMode value) => value switch
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs:109]: Could use a rebase; I think this method has changed in `main`

File: src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Courses/DeleteRow.cs [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            {
+                ProviderId = providerId
+            });
+
+            if (courseUpload == null)
+            {
+                return new NotFound();
+            }
+
+            var venueUploadRows = await _sqlQueryDispatcher.ExecuteQuery(new GetCourseUploadRows()
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Courses/DeleteRow.cs:81]:
```suggestion
            var courseUploadRows = await _sqlQueryDispatcher.ExecuteQuery(new GetCourseUploadRows()
```

**Review by gunndabad — APPROVED** [review]

**Review by soharadesign — APPROVED** [review]

File: src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs (hunk 3 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.Core/DataManagement/ParsedCsvCourseRow.cs:109]: Yeah those are in the other course download PR that hasn't been merged yet. Hima is currently testing it

File: src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Courses/DeleteRow.cs (hunk 4 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Courses/DeleteRow.cs]: done

File: src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.Courses.cs (hunk 2 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.Courses.cs]: done

File: src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.Courses.cs (hunk 1 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.Courses.cs:22]: done

**Pull request merged**